zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -200,6 +261,32 @@ private void init() throws InterpreterException, IOException, RepositoryExceptio
     }
   }
 
+  private boolean registerInterpreters(List<RegisteredInterpreter> registeredInterpreters,
+                                         String absolutePath) {
+    if (validateRegisterInterpreters(registeredInterpreters)) {
+      for (RegisteredInterpreter registeredInterpreter : registeredInterpreters) {
+        registeredInterpreter.setPath(absolutePath);
+        Interpreter.register(registeredInterpreter);
+      }
+      return true;
+    } else {
+      return false;
```

Comment:
It is only because of logging. :-) I'll think of it more fancy way with extracting `init()` method.